### DISCUSSION TITLE
Implement Claude Artifacts Functionality in Web Interface

### DISCUSSION THREAD

**Discussion by edisonzf2020** [open-webui/open-webui#3487] [Ideas]
Description:
We propose integrating Claude's Artifacts functionality into our web-based interface. Artifacts are a powerful feature that allows Claude to create and reference substantial, self-contained content during conversations. This feature enhances the user experience by providing a dedicated space for complex information, code snippets, or other content that benefits from separate display.
Key Features:

Artifact Creation: Enable Claude to generate artifacts within the web interface.
Artifact Display: Implement a separate UI window or panel to display artifacts.
Artifact Types Support: Support various artifact types, including:

Code (with syntax highlighting)
Markdown documents
HTML content
SVG images
Mermaid diagrams
React components

Artifact Referencing: Allow Claude to reference and update existing artifacts.
User Interaction: Provide users with the ability to view, copy, and potentially edit artifacts.

Benefits:

Enhanced Content Organization: Separates complex or lengthy content from the main conversation flow.
Improved Readability: Dedicated space for code, diagrams, and other structured content.
Increased Functionality: Enables Claude to provide more comprehensive and interactive responses.
Better User Experience: Allows users to focus on specific pieces of information without scrolling through the entire conversation.

Technical Considerations:

Integration with existing Claude API to support artifact creation and management.
Implementation of a flexible UI component to display various artifact types.
Ensuring proper rendering and functionality of different artifact types (e.g., SVG rendering, code syntax highlighting).
Consideration of performance impact when handling multiple or complex artifacts.

User Impact:
This feature will significantly enhance the capabilities of our web interface, allowing users to engage with Claude in more sophisticated and productive ways. It will be particularly beneficial for tasks involving coding, data analysis, creative writing, and complex problem-solving.
Next Steps:

Assess technical feasibility and resource requirements.
Create a detailed implementation plan.
Design UI mockups for artifact display and interaction.
Develop a prototype for initial testing and feedback.

By implementing the Artifacts functionality, we can provide a more powerful and versatile Claude experience to our web users, setting our platform apart from competitors and enabling more advanced use cases.

**Comment by AlgorithmicKing737** [comment]
Absolutely yes

**Reply by winchanter85** [reply]
tely！

**Comment by delta-whiplash** [comment]
We need this incredible feature

**Comment by abipalli** [comment]
Great way to digest relevant output without sacrificing response context

**Comment by edisonzf2020** [comment]
https://github.com/e2b-dev/e2b-cookbook/tree/main/examples/anthropic-power-artifacts

**Reply by MarcoWel** [reply]
> https://github.com/e2b-dev/e2b-cookbook/tree/main/examples/anthropic-power-artifacts

This is awesome!

However as a first step, just having "simple" code output on a side panel would be a great improvement in usability (compared to traditional "in-conversation" code display).

**Comment by andreybond** [comment]
YES!!! And maybe would be nice to make that artifact rendering region extendable via some sort of plugins. For example, regular  images (JPG?), or render (presentation page or LaTeX page/document) produced by corresponding code.

**Comment by bleetube** [comment]
Not sure if this is what is in the e2b-cookbook, but the Artifacts feature in Claude turns out to be part of the system prompt. The webui handles the rendering (and hiding of internal thinking about when to use artifacts). 

The prompt was disclosed here:
https://x.com/elder_plinius/status/1804052791259717665

**Reply by ashutoshpw** [reply]
We can add an option in the User settings to enable 'Artifacts' as a beta feature. If this option is enabled, we can include a prompt for artifacts within the system.

**Comment by tjbck** [comment]
WIP, I appreciate your patience!

**Reply by delta-whiplash** [reply]
Hello do you have any updates about this feature implementation ?

**Comment by thiswillbeyourgithub** [comment]
Just an idea, in a naive implementation artifacts are limited in size to the llm's max completion size. Maybe embedding a minimal [aider](aider.chat) to handle the code edition would be a **HUGE** improvement at minimal cost? This way we could also benefit from aider's author's meticulous testing of the specific type of editing format that works best for each llm. And FYI [aider can be scripted](https://aider.chat/docs/scripting.html)

**Comment by Pudochu** [comment]
Any information on the latest situation?
